snyk/README.md

```diff
@@ -1,43 +1,59 @@
 # Use Snyk to test your configuration files
 
 Author: [Shai Mendel](https://github.com/shaimendel)
-* Additional contributor: [Jim Armstrong](https://github.com/jimcodified)
+* Additional contributors: 
+	[Jim Armstrong](https://github.com/jimcodified)
+	[Matt Jarvis](https://github.com/mattj-io)
 
 [Snyk](https://https://snyk.io/) is a product to help developers to continuously find and fix vulnerabilities in open source libraries, containers and Infrastructure as Code configurations.
 
+## Requirements
+
+* The `snyk` binary must be on your path. See https://support.snyk.io/hc/en-us/articles/360003812538-Install-the-Snyk-CLI for details on installing, and register for a free Snyk account at https://app.snyk.io/login
+
 ## Usage
 
-The `snyk` extension can be run like so:
+The extension takes a number of arguments:
+
+name: a name for the resource. Useful for labeling when running multiple tests. Defaults to 'snyk'
+target: path to the file/dir or container image name:tag to test
+test_type: one of 'oss', 'container', or 'iac' corresponding to the various Snyk CLI tests
+test_deps: automatically set for test types other than container, allows for setting an external file/dir dependency
+extra_opts: additional CLI options; appended to the snyk command. see `snyk --help` for usage.
+          - when using the container switch this is where to insert the '--file=path/to/Dockerfile' option
+trigger: trigger mode - auto or manual, defaults to manual
+mode: control Snyk exit code, info will always exit 0 - gate or info, defaults to gate
```

Comment:
"gate" feels a little weird here - it's not actually gating anything, right?
IIUC, snyk will run after the code has already potentially been built and deployed, and this just controls whether the snyk resource shows red or green in Tilt when snyk has found a problem.

And if it's not actually gating, I'm not sure under what situation I'd want to use `mode=='info'` - I could see wanting to semi-disable snyk while trying to get something to work. Is there a circumstance under which one would want to configure a Tiltfile so that a failing scan always turns up as green in Tilt and makes it easier to miss? I figure if the user is uninterested in scan results, they can just choose not to look at them.

None of this is blocking for this PR - I just wanted to check if this made sense.

Reply:
I had the same thoughts, but not sure the right language so open to suggestions. From a local workflow perspective, I want to know that something isn't right, but not sure I want a big red flag. If we exit non-zero the Tilt output is build failed, which feels a lot like psychologically gating right ?

Reply:
@shaimendel is already aware of this PR, I think we're good for the minute